### PR TITLE
Removed none in style.display (not supported by IE11), replaced by em…

### PR DESCRIPTION
**Pull request by ciss1995**
DE31446. Removed none in style.display (not supported by IE11), replaced by em…
style.dislay = 'none' was causing the bug in IE11. I found out that style.display = 'none' attribute was not supported by IE11. I replaced it with the empty quotes, which reset the display to default. I tested it on Firefox, IE11, Chrome, and Edge through the Demo page as well as on my local instance.

### CODE REVIEW

**Comment by dbatiste** [comment]
Hey Cheick, it seems like there's something else going on here.  Removing the `display: none` from the collapsed element means the element will still be rendered, but overflowing, which makes me a bit nervous since there could be some strange behavior if the user is tabbing through the page.  I am suspicious of what's being assigned to the display property when expanding if no display property was being explicitly set before collapsing.

**Comment by ciss1995** [comment]
> Hey Cheick, it seems like there's something else going on here. Removing the `display: none` from the collapsed element means the element will still be rendered, but overflowing, which makes me a bit nervous since there could be some strange behavior if the user is tabbing through the page. I am suspicious of what's being assigned to the display property when expanding if no display property was being explicitly set before collapsing.

Another solution that I discovered was to remove the if statement! It seems like the 'none' display was carrying on to expand through data-d2l-ec-display, which Chrome/Firefox/Edge ignores but not IE11 causing the display = 'none' to be live in expand.

**Comment by dbatiste** [comment]
Hey Cheick, it looks like null is assigned to the `display` property on line 34:
```
node.style.display = node.getAttribute('data-d2l-ec-display');
```
And for IE11, it's resulting in unexpected value for the `scrollHeight` (0) measurement on line 40, resulting in no expansion.  The purpose of this measurement is to get an explicit height for the element for the purpose of animation.

What if we update like #34 to be:
```
var originalDisplay = (node.getAttribute('data-d2l-ec-display')) 
     ? !node.getAttribute('data-d2l-ec-display') : '';
node.style.display = originalDisplay;
```
For the case where the value is `null`, it would use `''` instead, and I _think_ this should result in the browser using the default display value for the element.

**Comment by dbatiste** [comment]
You could also try `unset`.  One key scenario to be careful of would be if the consumer has specified `display: inline-block;` in a CSS (not inline style), since this is a valid usage.

**Comment by ciss1995** [comment]
> Hey Cheick, it looks like null is assigned to the `display` property on line 34:
> 
> ```
> node.style.display = node.getAttribute('data-d2l-ec-display');
> ```
> And for IE11, it's resulting in unexpected value for the `scrollHeight` (0) measurement on line 40, resulting in no expansion. The purpose of this measurement is to get an explicit height for the element for the purpose of animation.
> 
> What if we update like #34 to be:
> 
> ```
> var originalDisplay = (node.getAttribute('data-d2l-ec-display')) 
>      ? !node.getAttribute('data-d2l-ec-display') : '';
> node.style.display = originalDisplay;
> ```
> For the case where the value is `null`, it would use `''` instead, and I _think_ this should result in the browser using the default display value for the element.

Hey Dave! I completely agree IE11 is resulting in a srollHeight of 0 due to the display result from line 34.
It seems like in Firefox/Chrome/Edge `node.style.display` takes the value from `node.getAttribute('data-d2l-ec-display') (null)` then outputs `''`, where in IE11, `node.style.display` outputs `none`. 
While the solution you provided works (with a slight delay upon expanding in IE11), I believe that the reason the value is null is that  `original.display` is never set to `data-d2l-ec-display` (because of the if statement). Any idea why we are doing that and is it necessary?

**Comment by dbatiste** [comment]
> While the solution you provided works (with a slight delay upon expanding in IE11), I believe that the reason the value is null is that original.display is never set to data-d2l-ec-display (because of the if statement). Any idea why we are doing that and is it necessary?

The intent of the `data-d2l-ec-display` is to store the original value so it can be restored upon expand, but only if it was being specified as an inline style.  We don't want to end up with inline styles applied (upon collapsing and expanding) if there wasn't originally because they'd override properties that might be defined in a stylesheet.  We only want to override the style temporarily for the purpose of the expand/collapse/behavior - we return the element back to it's original state.

**Comment by ciss1995** [comment]
> > While the solution you provided works (with a slight delay upon expanding in IE11), I believe that the reason the value is null is that original.display is never set to data-d2l-ec-display (because of the if statement). Any idea why we are doing that and is it necessary?
> 
> The intent of the `data-d2l-ec-display` is to store the original value so it can be restored upon expand, but only if it was being specified as an inline style. We don't want to end up with inline styles applied (upon collapsing and expanding) if there wasn't originally because they'd override properties that might be defined in a stylesheet. We only want to override the style temporarily for the purpose of the expand/collapse/behavior - we return the element back to it's original state.

Oh okay! Makes more sense! Thanks Dave!
I'll update the PR.

**Comment by ciss1995** [comment]
> > While the solution you provided works (with a slight delay upon expanding in IE11), I believe that the reason the value is null is that original.display is never set to data-d2l-ec-display (because of the if statement). Any idea why we are doing that and is it necessary?
> 
> The intent of the `data-d2l-ec-display` is to store the original value so it can be restored upon expand, but only if it was being specified as an inline style. We don't want to end up with inline styles applied (upon collapsing and expanding) if there wasn't originally because they'd override properties that might be defined in a stylesheet. We only want to override the style temporarily for the purpose of the expand/collapse/behavior - we return the element back to it's original state.

Oh okay! Makes more sense! Thanks Dave!
I'll update the PR.

File: d2l-dom-expand-collapse.html [hunk 1]
```diff
@@ -31,7 +31,7 @@
 					node.style.overflow = 'hidden';
 					node.style.transition = '';
 					node.style.height = '0px';
-					node.style.display = node.getAttribute('data-d2l-ec-display');
+					node.style.display = node.getAttribute('data-d2l-ec-display') ? !node.getAttribute('data-d2l-ec-display') : '';
```
Comment (dbatiste) [inline d2l-dom-expand-collapse.html]: I think I had a copy/paste error in my original suggestion.  I think you want:
```
node.style.display = node.getAttribute('data-d2l-ec-display') ? node.getAttribute('data-d2l-ec-display') : '';
```
i.e. remove the `!` from the 2nd argument.

Comment (ciss1995) [reply d2l-dom-expand-collapse.html]: Oh yes right, my mistake too! If `node.getAttribute('data-d2l-ec-display')` is not null we want it!

**Comment by dbatiste** [comment]
I guess the last thing would be - I think we want this applied to the `hybrid` branch instead of `master`.  And then we rerun the modularizer (I can probably help with that) and merge those changes to `master`.  That's probably why there are conflicts that need resolving... the HTML file does not exist in the P3 version.

**Pull request closed without merging**